### ISSUE TITLE
Allow per-window state

### ISSUE THREAD

**Issue by jidanni** [ibus/ibus#2100]
Another wish list item is "per-window state".

Currently "selecting next input method" affects **all** x windows.

Allowing a new "per-window state" would enable the  user to have different input methods independently in each x window!

Often there is no reason for the user to "select next input method" simultaneously  in all his  x windows. So with this new "per-window state" option enabled, finally he could have Input method X in window A, and input method Y on in window B!

**Comment by fujiwarat** [comment]
You could run ibus-setup

**Comment by jidanni** [comment]
OK, works good! (The user should be sure to restart nodm, xdm, etc. after changing this!)
By the way
 
      Share the same input method among all applications
              If this option is disabled, You can choose an input **metod** engine by input context.
Has a misspelling, and if it mentioned "all windows" along with / instead of "all applications" I would have tried it earlier.

**Comment by jidanni** [comment]
(OK after using that for one day, I am going back.

After a while Ibus gets totally confused about which (sub)window it is controlling.

Icons showing the method is enabled don't actually mean the method is in
effect for that (sub)window.

* Cannot correctly control popup windows. E.g., in Google Earth Pro I
was entering the name of a placemark and couldn't do it in Chinese.
Yes, the icons showed I had switched to Chewing mode, but it didn't
affect the field I was typing in.
* Can't even turn on the icon with the control sequence in some xterm
windows. Confusion between the control sequence and using the mouse on
the icon.
* In summary: one big mess. At least using the global setting it works
though not as convenient.

Yes all this is too complicated to make a reproducible bug report, so
never mind. Thanks anyway.)
